### PR TITLE
Feat: add search combine udt api(issue-1161)

### PR DESCRIPTION
**Pull request by Naupio**
## What problem does this PR solve?
link issue: #1161 
add combine search udt graphql api
## Check List
- search_udt
#### Test
- [x] unit test
#### Task
 - none

### CODE REVIEW

File: lib/godwoken_explorer/block.ex [hunk 1]
```diff
@@ -177,9 +177,22 @@ defmodule GodwokenExplorer.Block do
   end
 
   def update_blocks_finalized(latest_finalized_block_number) do
+    last_finalized_block_number =
```
Comment (zmcNotafraid) [inline lib/godwoken_explorer/block.ex:180]: 需要rebase，和你的代码无关

**Review by zmcNotafraid — CHANGES_REQUESTED** [review]

**Comment by FrederLu** [comment]
@Naupio 
Now "type": "NATIVE" in the interface, but it seems that ERC20/ERC72/ERC1555 needs to be displayed. Do you want to obtain relevant information through other interfaces?
image
image

File: lib/godwoken_explorer/block.ex (hunk 1 above)
Comment (Naupio) [reply lib/godwoken_explorer/block.ex]: fixed

**Comment by Naupio** [comment]
> @Naupio 1、Now "type": "NATIVE" in the interface, but it seems that ERC20/ERC721/ERC1555 needs to be displayed. Do you want to obtain relevant information through other interfaces? image image
> 
> 2、Prompt 500 with cursor request. image image

add eth_type now
```graphql
query {
  search_udt(input: { fuzzy_name: "%ERC%", limit: 1 }) {
    entries {
      id
      icon
      name
      symbol
      type
      eth_type
      contract_address_hash
    }
    metadata {
      total_count
      before
      after
    }
  }
}
```
```json
{
  "data": {
    "search_udt": {
      "entries": [
        {
          "contract_address_hash": "0xd8114ef1af2b1b4f9953ae764977c7c4ed409ea1",
          "eth_type": "ERC20",
          "icon": null,
          "id": 89222,
          "name": "ERC20Demo",
          "symbol": "ERC",
          "type": "NATIVE"
        }
      ],
      "metadata": {
        "after": "g3QAAAABaAJkAARkZXNjZAACaWRiAAFchg==",
        "before": null,
        "total_count": 720
      }
    }
  }
}
```

**Review by zmcNotafraid — APPROVED** [review]

**Comment by FrederLu** [comment]
@Naupio 
`eth_type` has been added, but the request with a cursor still prompts 500.

**Comment by Naupio** [comment]
> @Naupio `eth_type` has been added, but the request with a cursor still prompts 500.

fixed and added more test case

**Review by FrederLu — APPROVED** [review]

**Pull request merged**